Improve moduleloader

danielkesselberg opened the pull request:
**Dangerous change:** Removed conditional caching from ModuleAutoLoader

githubjeka commented on protected/humhub/components/bootstrap/ModuleAutoLoader.php:
```diff
@@ -43,11 +62,52 @@ public function bootstrap($app)
+        };
+
+        try {
+            return FileHelper::findDirectories(Yii::getAlias($path, true), ['filter' => $hasConfigurationFile]);
```
https://www.yiiframework.com/doc/api/2.0/yii-helpers-basefilehelper#findDirectories()-detail

What about the case when

- moduleDir
   - tests
      - config.php
   - config.php

danielkesselberg replied:
Good catch. I will look into recursive true/false tomorrow.

githubjeka commented on protected/humhub/components/bootstrap/ModuleAutoLoader.php:
```diff
@@ -43,11 +62,52 @@ public function bootstrap($app)
                     }
                 }
             }
-            if (!YII_DEBUG) {
```
Can you describe what's wrong with this line?

githubjeka replied:
> Conditional caching could lead to unexpected results when YII_DEBUG is
enabled while cache is not DummyCache

Ah, ok

githubjeka commented on protected/humhub/components/bootstrap/ModuleAutoLoader.php:
```diff
@@ -19,8 +20,26 @@
+
+    /**
+     * Find available modules
+     * @deprecated 1.3 replace call for locateModules with findModules and handle caching outside of method (e.g. in boostrap)
```
How about remove `@deprecated` and rewrite it by `findModules()`, now.

danielkesselberg commented:
@luke- Code Climate not happy but build is ok.

acs-ferreira commented:
@danielkesselberg I believe Codeclimate can be fixed using `getOrSet()` here.

danielkesselberg commented:
@acs-ferreira CodeClimate is unhappy because locateModules is way to complex. There is nothing we can do right now. I added `@depcreated` and added a cleaner version with some unit tests.
